README.md

```diff
@@ -1,2 +1,313 @@
+        if votesReceived > (|nodes| + 1) / 2 then
+            currentRole := leader
+            currentLeader := nodeld
+            cancel election timer
+            for each follower in nodes \ {nodeld} do
+                sentLength[follower] := log.length
+                ackedLength[follower] := 0
+                REPLICATELOG(nodeld, follower)
+            end for
+        end if
+    else if term > currentTerm then
+        currentTerm := term
+        currentRole := follower
+        votedFor := null
+        cancel election timer
+    end if
+end on
+
+```
+
+This phase handles the response to a vote request from a candidate.
+
+* If the node is a candidate and receives a vote response for the current term, it adds the vote to the received votes.
+* If the candidate receives votes from a majority of the nodes, it becomes the leader.
+* If the received term is greater than the current term, the node updates its term and role.
+
+## Broadcasting Messages
+```shell
+on request to broadcast msg at node nodeld do
+    if currentRole = leader then
+        append the record (msg: msg, term: currentTerm) to log
+        ackedLength[nodeld] := log.length
+        for each follower in nodes \ {nodeld} do
+            REPLICATELOG(nodeld, follower)
+        end for
+    else
+        forward the request to currentLeader via a FIFO link
+    end if
+end on
+
+periodically at node nodeld do
+    if currentRole = leader then
+        for each follower in nodes \ {nodeld} do
+            REPLICATELOG(nodeld, follower)
+        end for
+    end if
+end on
+
+```
+
+This phase handles broadcasting messages in the cluster.
+
+* If the node is the leader, it appends the message to its log and replicates it to the followers.
+* If the node is not the leader, it forwards the request to the current leader.
+
+## Replicating from Leader to Followers
+
+```shell
+function REPLICATELOG(leaderId, followerId)
+    prefixLen := sentLength[followerId]
+    suffix := (log[prefixLen+1:], ...)
+    prefixTerm := 0
+    if prefixLen > 0 then
+        prefixTerm := log[prefixLen].term
+    end if
+    send (LogRequest, leaderId, currentTerm, prefixLen, prefixTerm, commitLength, suffix) to followerId
+end function
+
+```
+This phase handles log replication from the leader to its followers.
+
+* The `REPLICATELOG` function iscalled when there are new log entries or periodically.
+* The function determines the prefix and suffix of the log entries to send to a follower.
+* A `LogRequest` message is sent to the follower containing the necessary log information.
+
+## Followers Receiving Messages
+
+```shell
+on receiving (LogRequest, leaderId, term, prefixLen, prefixTerm, leaderCommit, suffix) at node nodeld do
+    if term > currentTerm then
+        currentTerm := term
+        currentRole := follower
+        currentLeader := leaderId
+    end if
+    logOk := (log.length >= prefixLen) and (prefixLen = 0 or log[prefixLen].term = prefixTerm)
```

Comment:
In here, I think it should be "logOk := (log.length ≥ prefixLen) ∧
(prefixLen = 0 ∨ log[prefixLen − 1].term = prefixTerm)" instead of "logOk := (log.length ≥ prefixLen) ∧
(prefixLen = 0 ∨ log[prefixLen].term = prefixTerm)".